clients/go/compat.go

```diff
@@ -0,0 +1,306 @@
+	return w.SignFn(digest)
+}
+
+func (b WrappedBackend) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	header, err := b.Backend.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	blockHash := header.Hash()
+	leash := NewLeash(header.Nonce.Uint64(), header.Number.Uint64(), blockHash[:], DefaultBlockRange)
+	publicKey, _ := tweetnacl.ScalarMultBase(crypto.FromECDSA(&b.Key))
+	keypair := tweetnacl.KeyPair{
+		PublicKey: publicKey,
+		SecretKey: crypto.FromECDSA(&b.Key),
+	}
+	cipher, _ := NewX255919DeoxysIICipher(keypair, b.RuntimePublicKey)
+	dataPack, _ := NewDataPack(b.Signer, tx.ChainId().Uint64(), b.TransactOpts.From[:], addressToByte(tx.To()), tx.Gas(), tx.GasPrice(), tx.Value(), tx.Data(), leash)
+
+	legacyTx := &types.LegacyTx{
+		To:       tx.To(),
+		Nonce:    tx.Nonce(),
+		GasPrice: tx.GasPrice(),
+		Gas:      DefaultGasLimit,
+		Value:    tx.Value(),
+		Data:     dataPack.EncryptEncode(cipher),
+	}
+
+	baseTx := *types.NewTx(legacyTx)
+
+	signer := types.LatestSignerForChainID(tx.ChainId())
+	signature, _ := crypto.Sign(signer.Hash(&baseTx).Bytes(), &b.Key)
+	signedTx, _ := baseTx.WithSignature(signer, signature)
+
+	return b.Backend.SendTransaction(ctx, signedTx)
+}
+
+func (b WrappedBackend) CallContract(ctx context.Context, call ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
+	header, err := b.Backend.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	blockHash := header.Hash()
+	leash := NewLeash(header.Nonce.Uint64(), header.Number.Uint64(), blockHash[:], DefaultBlockRange)
+
+	publicKey, _ := tweetnacl.ScalarMultBase(crypto.FromECDSA(&b.Key))
+	keypair := tweetnacl.KeyPair{
+		PublicKey: publicKey,
+		SecretKey: crypto.FromECDSA(&b.Key),
+	}
+	cipher, _ := NewX255919DeoxysIICipher(keypair, b.RuntimePublicKey)
+	dataPack, err := NewDataPack(b.Signer, b.ChainID.Uint64(), call.From[:], addressToByte(call.To), DefaultGasLimit, call.GasPrice, call.Value, call.Data, leash)
+
+	if err != nil {
+		return nil, err
+	}
+
+	call.Data = dataPack.EncryptEncode(cipher)
+	res, err := b.Backend.CallContract(ctx, call, blockNumber)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return cipher.DecryptEncoded(res)
+}
+
+func (b WrappedBackend) CodeAt(ctx context.Context, contract common.Address, blockNumber *big.Int) ([]byte, error) {
+	return b.Backend.CodeAt(ctx, contract, blockNumber)
+}
+
+func (b WrappedBackend) EstimateGas(ctx context.Context, call ethereum.CallMsg) (gas uint64, err error) {
+	header, err := b.Backend.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	blockHash := header.Hash()
+	leash := NewLeash(header.Nonce.Uint64(), header.Number.Uint64(), blockHash[:], DefaultBlockRange)
+
+	dataPack, _ := NewDataPack(b.Signer, b.ChainID.Uint64(), call.From[:], addressToByte(call.To), DefaultGasLimit, call.GasPrice, call.Value, call.Data, leash)
+	call.Data = dataPack.Encode()
+
+	return b.Backend.EstimateGas(ctx, call)
```

Comment:
For this one, you're just going to have to return a constant big number, as sapphire doesn't support gas estimation at the moment. You can comment this part out, though. It'll be useful later.